test/rebar3_hex_integration_SUITE.erl

```diff
@@ -501,216 +510,296 @@ publish_multi_errors_test(Config) ->
+         },
+    #{rebar_state := State, repo := Repo} = setup_state(P, Config),
     create_user(?default_username, ?default_password, ?default_email, Repo),
-    Command = ["remove", "truecoat", "wade@foo.bar"],
-    RepoConfig = [{repos,[Repo]}],
-    {ok, RemoveState} = test_utils:mock_command(rebar3_hex_owner, Command, RepoConfig, State),
-
-    ?assertMatch({ok, RemoveState}, rebar3_hex_owner:do(RemoveState)).
+    ?assertMatch({ok, State}, rebar3_hex_owner:do(State)).
 
 owner_list_test(Config) ->
-    P = #{app => "valid", mocks => [owner]},
-    {ok, #{rebar_state := State, repo := Repo}} = setup_state(P, Config),
+    P = #{
+          command => #{provider => rebar3_hex_owner, args => ["list", "truecoat"]}, 
+          project => "valid", 
+          mocks => [owner]
+         },
+    #{rebar_state := State, repo := Repo} = setup_state(P, Config),
     create_user(?default_username, ?default_password, ?default_email, Repo),
-    Command = ["list", "truecoat"],
-    RepoConfig = [{repos,[Repo]}],
-    {ok, ListState} = test_utils:mock_command(rebar3_hex_owner, Command, RepoConfig, State),
-
-    ?assertMatch({ok, ListState}, rebar3_hex_owner:do(ListState)).
+    ?assertMatch({ok, State}, rebar3_hex_owner:do(State)).
 
 bad_command_test(Config) ->
-    P = #{app => "valid", mocks => [deauth]},
-    {ok, #{rebar_state := State, repo := Repo}} = setup_state(P, Config),
-    Command = ["bad_command"],
-    RepoConfig = [{repos,[Repo]}],
-    {ok, BadcommandState} = test_utils:mock_command(rebar3_hex_user, Command, RepoConfig, State),
-
-    ?assertThrow({error,{rebar3_hex_user,bad_command}}, rebar3_hex_user:do(BadcommandState)).
+    P = #{command => #{provider => rebar3_hex_user, args => ["bad_command"]}, project => "valid", mocks => [deauth]},
+    #{rebar_state := State} = setup_state(P, Config),
+    ?assertThrow({error,{rebar3_hex_user,bad_command}}, rebar3_hex_user:do(State)).
 
 revert_invalid_ver_test(Config) ->
-    P = #{app => "valid", mocks => [publish_revert], version => "eh?"},
-    {ok, #{rebar_state := State, repo := Repo}} = setup_state(P, Config),
-    RepoConfig = [{repos,[Repo]}],
-    {ok, RevState} = test_utils:mock_command(rebar3_hex_revert, ["valid", "eh?"], RepoConfig, State),
+    P = #{
+          command => #{provider => rebar3_hex_revert, args => ["valid", "eh?"]}, 
+          project => "valid", 
+          mocks => [publish_revert], 
+          version => "eh?"
+         },
+    #{rebar_state := State} = setup_state(P, Config),
+     ?assertThrow(rebar_abort, rebar3_hex_revert:do(State)).
 
-     ?assertThrow(rebar_abort, rebar3_hex_revert:do(RevState)).
 %%%%%%%%%%%%%%%%%%
 %%%  Helpers   %%%
 %%%%%%%%%%%%%%%%%%
 
-
 %% setup_state/2 takes a set of parameters and a CT config
 %% The with param is used to setup predefined mock scenarios. May be an empty list if no mocks are required.
 setup_state(P, Config) ->
     Params = maps:merge(?default_params, P),
-    #{app := AppName,
+    #{project := ProjectName,
```

Comment:
We now use project and app key parameters to so that we can support testing umbrellas.